docs: add docs to functions and setup auto-generation of docs (#70)
JoshuaKGoldberg/ts-api-utils: .github/DEVELOPMENT.md

This package includes several forms of linting to enforce consistent code quality and styling.
Each should be shown in VS Code, and can be run manually on the command-line:

> - `pnpm lint:docs` ([typedocs](https://typedoc.org/)): Checks that all the code is documented

Reviewer: [Nitpick]

```suggestion
- `pnpm lint:docs` ([typedocs](https://typedoc.org)): Checks that all the code is documented
```

- `pnpm lint:knip` ([knip](https://github.com/webpro/knip)): Detects unused files, dependencies, and code exports

> - `pnpm lint:knip:production` ([knip](https://github.com/webpro/knip)): The same as `lint:knip` but looks purely at production code/dependencies

Reviewer: Ah, I forgot about this one 😄

```

You should also see suggestions from TypeScript in your editor.

## Documentation

> You can generate documentation site with `pnpm docs`.

Reviewer: [Nitpick] Grammar, and a bit more info.

```suggestion
You can generate a [TypeDoc](https://typedoc.org) documentation site with `pnpm docs`.
```

> You can generate a [TypeDoc](https://typedoc.org) documentation site with `pnpm docs`.

Reviewer: On my Mac, this just opens the GitHub site. 

```suggestion
You can generate a [TypeDoc](https://typedoc.org) documentation site with `pnpm run docs`.
```

> To browse the docs locally you'll need a local webserver server (for VS Code users, we recommend [Live Preview](https://marketplace.visualstudio.com/items?itemName=ms-vscode.live-server)).

Reviewer: [Question] I haven't been using one - is there a real benefit to?

[Request] If not, how about:

```suggestion
To browse the docs locally, open `docs/dev/index.html` in a web browser.
```

Author: You can view the pure html of `docs/*/index.html` files with the `file://` protocol, but some features that rely on JavaScript aren't available with this protocol (due to CORS policy). One being the version selector.

`docs/index.html` itself isn't viewable with the `file://` protocol.

Reviewer: Ah, got it. I'd suggest adding in a `package.json` command like `"docs:serve": "http-serve docs/generated` then. But if you have a strong preference for this then I'm down to try it out 😄

Author: We can add that command if you like.
I was just using "Live Preview" because I already had that extension installed. You just have to right click on an html file and select "Show Preview".

> Once you've generated the docs, you can view them by first running `pnpm docs:server`, then browsering to <http://127.0.0.1:8080> (or similar - check the console output of the last command).

Reviewer: [Typo]

```suggestion
Once you've generated the docs, you can view them by first running `pnpm docs:serve`, then browsing to <http://127.0.0.1:8080> (or similar - check the console output of the last command).
```

> 

Reviewer: [Suggestion] I've found this to be helpful:

```suggestion

> Tip: Run `pnpm run docs --watch` to keep the generated docs up-to-date as you save files.

```
